### ISSUE TITLE
Handling a negative EXT-X-MEDIA-SEQUENCE

### ISSUE THREAD

**Issue by wlerin** [globocom/m3u8#188]
So, this is completely nonstandard (decimal-integer is defined to start at 0), but I have encountered a number of streams that provide a negative media-sequence in conjunction with EXT-X-DISCONTINUITY. Thanks to `normalize_attribute`, this library ends up throwing a ValueError when it tries to cast e.g. "_3" to an int. 

Is there any way to work around this? Or can the parser be changed to, well, *not* normalize EXT-X-MEDIA-SEQUENCE values? I'm not entirely sure why it does so currently.

**Comment by leandromoreira** [comment]
a negative media sequence is really not usual maybe you could pass your own parser, I think we build up something that enables its users to do that isn't that right @mauricioabreu ?

**Comment by mauricioabreu** [comment]
@wlerin @leandromoreira 
I think you can use the `custom_tags_parser` feature: https://github.com/globocom/m3u8#custom-tags but I am not 100% sure if it will work.

```python
import m3u8

def media_sequence(line, data, lineno):
    if line.startswith('#EXT-MEDIA-SEQUENCE:'):
        media_sequence = line.split(':')[1]
        data['media_sequence'] = abs(int(media_sequence.strip()))

m3u8_obj = m3u8.load('http://videoserver.com/playlist.m3u8', custom_tags_parser=media_sequence)
print(m3u8_obj.data['media_sequence'])
```

Anyways, I think it could be easier if we just call `abs` inside the m3u8 core. What do you think @leandromoreira ?

**Comment by wlerin** [comment]
```
    m3u8_obj = m3u8.load(src, headers=headers or DEFAULT_HEADERS)
  File "/home/wlerin/.virtualenvs/dev/lib/python3.7/site-packages/m3u8/__init__.py", line 56, in load
    return _load_from_uri(uri, timeout, headers, custom_tags_parser, verify_ssl)
  File "/home/wlerin/.virtualenvs/dev/lib/python3.7/site-packages/m3u8/__init__.py", line 74, in _load_from_uri
    return M3U8(content, base_uri=base_uri, custom_tags_parser=custom_tags_parser)
  File "/home/wlerin/.virtualenvs/dev/lib/python3.7/site-packages/m3u8/model.py", line 147, in __init__
    self.data = parse(content, strict, custom_tags_parser)
  File "/home/wlerin/.virtualenvs/dev/lib/python3.7/site-packages/m3u8/parser.py", line 81, in parse
    _parse_simple_parameter(line, data, int)
  File "/home/wlerin/.virtualenvs/dev/lib/python3.7/site-packages/m3u8/parser.py", line 334, in _parse_simple_parameter
    return _parse_and_set_simple_parameter_raw_value(line, data, cast_to, True)
  File "/home/wlerin/.virtualenvs/dev/lib/python3.7/site-packages/m3u8/parser.py", line 328, in _parse_and_set_simple_parameter_raw_value
    param, value = _parse_simple_parameter_raw_value(line, cast_to, normalize)
  File "/home/wlerin/.virtualenvs/dev/lib/python3.7/site-packages/m3u8/parser.py", line 324, in _parse_simple_parameter_raw_value
    return param, cast_to(value)
ValueError: invalid literal for int() with base 10: '_3'
```
I suppose I could patch in a try-except someone under `_parse_simple_parameter`

**Comment by leandromoreira** [comment]
> Anyways, I think it could be easier if we just call abs inside the m3u8 core. What do you think @leandromoreira ?

@mauricioabreu I don't think we should do it because it can hide some real world problems (like a negative media sequence) but what do you think @flavioribeiro @igorsobreira

**Comment by mauricioabreu** [comment]
@leandromoreira agreed!

**Comment by wlerin** [comment]
In the long run I'd like to see a solution to this in the m3u8 library--e.g. I don't think a value that should be an integer should be having - changed to _, even if a negative value isn't allowed by the specification. But for the meantime inserting this at the top of my code seems to resolve the issue:

```python
from m3u8 import parser

# Handle negative media-sequence values
# does this have any, ahem, negative consequences?
def _showroom_parse_simple_parameter_raw_value(line, cast_to=str, normalize=False):
    param, value = line.split(':', 1)
    param = parser.normalize_attribute(param.replace('#EXT-X-', ''))
    if normalize:
        value = parser.normalize_attribute(value)
    try:
        return param, cast_to(value)
    except ValueError:
        return param, cast_to(value.replace('_', '-'))

parser._parse_simple_parameter_raw_value = _showroom_parse_simple_parameter_raw_value
import m3u8
```

**Comment by mauricioabreu** [comment]
@wlerin can you check #189 ?

**Comment by wlerin** [comment]
@mauricioabreu That would solve my issue.

**Comment by wlerin** [comment]
Fixed with #189

**Comment by mauricioabreu** [comment]
@wlerin new release 0.5.4
